core/common/src/Instant.kt

```diff
@@ -25,12 +26,166 @@ import kotlin.time.*
  * corresponding to leap seconds, but instead "smears" positive and negative leap seconds among the last 1000 seconds
  * of the day when a leap second happens.
  *
- * Some ways in which [Instant] can be acquired are:
- * - [Clock.now] can be used to query the current moment for the given clock. With [Clock.System], it is the current
- *   moment as the platform sees it.
- * - [Instant.parse] parses an ISO-8601 string.
- * - [Instant.fromEpochMilliseconds] and [Instant.fromEpochSeconds] construct the instant values from the amount of time
- *   since `1970-01-01T00:00:00Z` (the Unix epoch).
+ * ### Obtaining the current moment
+ *
+ * The [Clock] interface is the primary way to obtain the current moment:
+ *
+ * ```
+ * val clock: Clock = Clock.System
+ * val instant = clock.now()
+ * ```
+ *
+ * The [Clock.System] implementation uses the platform-specific system clock to obtain the current moment.
+ * Note that this clock is not guaranteed to be monotonic, and it may be adjusted by the user or the system at any time,
+ * so it should not be used for measuring time intervals.
+ * For that, consider using [TimeSource.Monotonic] and [TimeMark] instead of [Clock.System] and [Instant].
+ *
+ * ### Obtaining human-readable representations
+ *
+ * #### Date and time
+ *
+ * [Instant] is essentially the number of seconds and nanoseconds since a designated moment in time,
+ * stored as something like `1709898983.123456789`.
+ * [Instant] contains no information about what day or time it is, as this depends on the time zone.
+ * To work with this information for a specific time zone, obtain a [LocalDateTime] using [Instant.toLocalDateTime]:
+ *
+ * ```
+ * val instant = Instant.fromEpochSeconds(1709898983, 123456789)
+ * instant.toLocalDateTime(TimeZone.of("Europe/Berlin")) // 2024-03-08T12:56:23.123456789
+ * instant.toLocalDateTime(TimeZone.UTC) // 2024-03-08T11:56:23.123456789
+ * ```
+ *
+ * For values very far in the past or the future, this conversion may fail.
+ * The specific range of values that can be converted to [LocalDateTime] is platform-specific, but at least
+ * [DISTANT_PAST], [DISTANT_FUTURE], and all values between them can be converted to [LocalDateTime].
```

Comment:
I'd like things like `a..b` to be between backticks so that `..` is not treated as just some broken punctuation, but then, we can't have references to entities.
@qwwdfsad, IIRC, you're collecting feature requests for the KDoc syntax—navigable links in code blocks and inline code would be nice.

Reply:
https://github.com/Kotlin/dokka/issues/3571

Don't hesitate to file it in Dokka with `kdoc-spec` label